src/libsync/syncengine.cpp

```diff
@@ -795,8 +795,56 @@ void SyncEngine::slotPropagationFinished(bool success)
+        const QString dataFingerprintOverrideHostHash = QStringLiteral("63debc9ef6d217649ea70632ca573a1db7a237ba61c48cdd2bf797f7060233db");
+        const auto accountHost = account()->url().host();
+        const auto accountDisplayName = account()->displayName();
+
+        if (_dataFingerprintSetFailCount >= 0) {
+            qCWarning(lcEngine) << "setDataFingerprint has failed for account" << accountDisplayName << "on host" << accountHost << "due to sync errors. Checking the possibility for override...";
+
+            if (_dataFingerprintSetFailCount > 0) {
+                if (_dataFingerprintSetFailCount >= dataFingerprintOverrideThreshold) {
+                    qCWarning(lcEngine) << "All sync attempts failed for account" << accountDisplayName << "on host" << accountHost << "setting the dataFingerprint anyway.";
+                    _journal->setDataFingerprint(_discoveryPhase->_dataFingerprint);
+                    // this mechanism should only run once per app launch
+                    _dataFingerprintSetFailCount = -1;
+                } else {
+                    ++_dataFingerprintSetFailCount;
+                    // request to start sync again as it won't happen by itself unless the file has changed on the server or in the local folder
+                    shouldStartSyncAgain = true;
+                }
+            } else {
+                // only compare hash once
+                // if it matches - we don't need to calculate it again while _dataFingerprintSetFailCount is greater than 0
+                const auto accountHostHash = QString::fromUtf8(QCryptographicHash::hash(accountHost.toUtf8(),QCryptographicHash::Sha256).toHex());
+
+                if (accountHostHash == dataFingerprintOverrideHostHash) {
+                    qCInfo(lcEngine) << "accountHostHash" << accountHostHash << "equals to dataFingerprintOverrideHostHash" << dataFingerprintOverrideHostHash << "_dataFingerprintSetFailCount" << _dataFingerprintSetFailCount;
+                    ++_dataFingerprintSetFailCount;
+                    // request to start sync again as it won't happen by itself unless the file has changed on the server or in the local folder
+                    shouldStartSyncAgain = true;
+                } else {
+                    qCInfo(lcEngine) << "accountHostHash" << accountHostHash << "differs from dataFingerprintOverrideHostHash" << dataFingerprintOverrideHostHash;
+                    // give up on calculating the has next time, as it's not the host we are looking for
+                    _dataFingerprintSetFailCount = -1;
+                }
+            }
+        } else {
+            qCWarning(lcEngine) << "setDataFingerprint was overridden already for account" << accountDisplayName << "on host" << accountHost << "but is failing again! Or, it's not the host that we are looking for.";
+        }
+    };
+    //
+
     if (success && _discoveryPhase) {
+        checkAndOverrideSetDataFingerprint();
```

Comment:
Don't you want the comment here as well? In a way I guess you'll just do a git revert?

Reply:
@er-vin Oops. Actually, I shouldn't call this lambda here, just in the else if statement. Forgot to remove it after testing that it works :)